File: skds/laya/ApolloSDK/ApolloClientConfig.ts
/**
 * Apollo 客户端配置
 */

/**
 * 日志级别
 */
export enum LogLevel {
    DEBUG = 0,
    INFO = 1,
    WARNING = 2,
    ERROR = 3,
    NONE = 4
}

/**
 * Apollo 客户端配置类
 */
export class ApolloClientConfig {
    /**
     * 服务器地址
     */
    public serverAddress: string = 'localhost';

    /**
     * 服务器端口
     */
    public port: number = 8080;

    /**
     * 是否使用 WebSocket Secure (wss://)
     */
    public useSSL: boolean = false;

    /**
     * 连接超时时间（毫秒）
     */
    public connectTimeout: number = 10000;

    /**
     * 心跳间隔（毫秒）
     */
    public heartbeatInterval: number = 30000;

    /**
     * 心跳超时次数阈值
     */
    public heartbeatTimeoutThreshold: number = 3;

    /**
     * 重连尝试次数
     */
    public reconnectAttempts: number = 3;

    /**
     * 重连间隔（毫秒）
     */
    public reconnectInterval: number = 5000;

    /**
     * 日志级别
     */
    public logLevel: LogLevel = LogLevel.INFO;

    /**
     * 是否启用自动重连
     */
    public enableAutoReconnect: boolean = true;

    /**
     * 消息压缩
     */
    public enableCompression: boolean = false;

    /**
     * 构造函数
     */
    constructor(options?: Partial<ApolloClientConfig>) {
        if (options) {
            for (const key in options) {
                if (options.hasOwnProperty(key)) {
                    this[key] = options[key];
                }
            }
        }
    }

    /**
     * 获取 WebSocket URL
     */
    public getWebSocketURL(): string {
        const protocol = this.useSSL ? 'wss://' : 'ws://';
        return `${protocol}${this.serverAddress}:${this.port}`;
    }

    /**
     * 创建默认配置
     */
    public static createDefault(): ApolloClientConfig {
        return new ApolloClientConfig();
    }

    /**
     * 创建生产环境配置
     */
    public static createProduction(serverAddress: string, port: number): ApolloClientConfig {
        return new ApolloClientConfig({
            serverAddress: serverAddress,
            port: port,
            useSSL: true,
            logLevel: LogLevel.WARNING,
            enableAutoReconnect: true
        });
    }

    /**
     * 创建开发环境配置
     */
    public static createDevelopment(serverAddress: string, port: number): ApolloClientConfig {
        return new ApolloClientConfig({
            serverAddress: serverAddress,
            port: port,
            useSSL: false,
            logLevel: LogLevel.DEBUG,
            enableAutoReconnect: true
        });
    }
}

File: skds/unity/ApolloSDK/ApolloClientConfig.cs
using System;

namespace ApolloSDK
{
    /// <summary>
    /// 传输协议类型
    /// </summary>
    public enum TransportProtocol
    {
        Tcp,
        WebSocket,
        Kcp
    }

    /// <summary>
    /// 加密模式
    /// </summary>
    public enum EncryptionMode
    {
        None,
        Aes128,
        Aes256,
        ChaCha20
    }

    /// <summary>
    /// 日志级别
    /// </summary>
    public enum LogLevel
    {
        Error,
        Warning,
        Info,
        Debug,
        Trace
    }

    /// <summary>
    /// Apollo客户端配置
    /// </summary>
    public class ApolloClientConfig
    {
        /// <summary>
        /// 服务器地址
        /// </summary>
        public string ServerAddress { get; set; } = "localhost";

        /// <summary>
        /// 服务器端口
        /// </summary>
        public int Port { get; set; } = 7700;

        /// <summary>
        /// 传输协议
        /// </summary>
        public TransportProtocol Protocol { get; set; } = TransportProtocol.Tcp;

        /// <summary>
        /// 心跳间隔（秒）
        /// </summary>
        public int HeartbeatInterval { get; set; } = 15;

        /// <summary>
        /// 心跳超时时间（秒）
        /// </summary>
        public int HeartbeatTimeout { get; set; } = 45;

        /// <summary>
        /// 连接超时时间（毫秒）
        /// </summary>
        public int ConnectionTimeout { get; set; } = 10000;

        /// <summary>
        /// 加密模式
        /// </summary>
        public EncryptionMode Encryption { get; set; } = EncryptionMode.Aes256;

        /// <summary>
        /// 加密密钥
        /// </summary>
        public string EncryptionKey { get; set; } = "";

        /// <summary>
        /// 资源槽位（用于灰度发布）
        /// </summary>
        public string ResourceSlot { get; set; } = "A";

        /// <summary>
        /// 日志级别
        /// </summary>
        public LogLevel LogLevel { get; set; } = LogLevel.Info;

        /// <summary>
        /// 日志文件路径
        /// </summary>
        public string LogFilePath { get; set; } = "apollo_sdk.log";

        /// <summary>
        /// 是否启用日志上传
        /// </summary>
        public bool EnableLogUpload { get; set; } = false;

        /// <summary>
        /// 日志上传地址
        /// </summary>
        public string LogUploadUrl { get; set; } = "";

        /// <summary>
        /// 最大重连次数
        /// </summary>
        public int MaxReconnectAttempts { get; set; } = 5;

        /// <summary>
        /// 重连延迟（毫秒）
        /// </summary>
        public int ReconnectDelay { get; set; } = 3000;

        /// <summary>
        /// 是否启用指数退避重连
        /// </summary>
        public bool UseExponentialBackoff { get; set; } = true;

        /// <summary>
        /// 消息压缩阈值（字节）
        /// </summary>
        public int CompressionThreshold { get; set; } = 1024;

        /// <summary>
        /// 是否启用消息压缩
        /// </summary>
        public bool EnableCompression { get; set; } = true;

        /// <summary>
        /// 接收缓冲区大小
        /// </summary>
        public int ReceiveBufferSize { get; set; } = 65536;

        /// <summary>
        /// 发送缓冲区大小
        /// </summary>
        public int SendBufferSize { get; set; } = 65536;

        /// <summary>
        /// 是否启用诊断模式
        /// </summary>
        public bool EnableDiagnostics { get; set; } = false;

        /// <summary>
        /// 诊断消息缓存数量
        /// </summary>
        public int DiagnosticMessageCache { get; set; } = 1000;

        /// <summary>
        /// 验证配置有效性
        /// </summary>
        public void Validate()
        {
            if (string.IsNullOrEmpty(ServerAddress))
            {
                throw new ArgumentException("ServerAddress is required");
            }

            if (Port <= 0 || Port > 65535)
            {
                throw new ArgumentException("Port must be between 1 and 65535");
            }

            if (HeartbeatInterval <= 0)
            {
                throw new ArgumentException("HeartbeatInterval must be greater than 0");
            }

            if (HeartbeatTimeout < HeartbeatInterval)
            {
                throw new ArgumentException("HeartbeatTimeout must be greater than or equal to HeartbeatInterval");
            }

            if (ConnectionTimeout <= 0)
            {
                throw new ArgumentException("ConnectionTimeout must be greater than 0");
            }

            if (ReceiveBufferSize <= 0)
            {
                throw new ArgumentException("ReceiveBufferSize must be greater than 0");
            }

            if (SendBufferSize <= 0)
            {
                throw new ArgumentException("SendBufferSize must be greater than 0");
            }

            if (Encryption != EncryptionMode.None && string.IsNullOrEmpty(EncryptionKey))
            {
                throw new ArgumentException("EncryptionKey is required when encryption is enabled");
            }
        }

        /// <summary>
        /// 创建默认配置
        /// </summary>
        /// <returns></returns>
        public static ApolloClientConfig CreateDefault()
        {
            return new ApolloClientConfig();
        }

        /// <summary>
        /// 从配置文件加载
        /// </summary>
        /// <param name="configPath">配置文件路径</param>
        /// <returns></returns>
        public static ApolloClientConfig LoadFromFile(string configPath)
        {
            // TODO: 实现从JSON文件加载配置
            // 可以使用Newtonsoft.Json或Unity的JsonUtility
            return CreateDefault();
        }

        /// <summary>
        /// 保存配置到文件
        /// </summary>
        /// <param name="configPath">配置文件路径</param>
        public void SaveToFile(string configPath)
        {
            // TODO: 实现保存到JSON文件
            // 可以使用Newtonsoft.Json或Unity的JsonUtility
        }
    }
}